launcher/minecraft/Rule.cpp

```diff
@@ -58,12 +58,11 @@ QList<std::shared_ptr<Rule>> rulesFromJsonV4(const QJsonObject &objectWithRules)
 
         auto osObj = osVal.toObject();
         auto osNameVal = osObj.value("name");
-        if (!osNameVal.isString())
+        auto osArchVal = osObj.value("arch");
+        if (!osNameVal.isString() && !osArchVal.isString())
```

Comment:
```suggestion
        if (!osNameVal.isString() || !osArchVal.isString())
```

Reply:
I don't think this is it. The rule is correct if either arch exists or name exists.